pipenv/routines/uninstall.py

```diff
@@ -82,22 +82,32 @@ def do_uninstall(
     if all:
         click.secho(
             click.style(
-                "Un-installing all {}...".format(click.style("[packages]", fg="yellow")),
+                "Un-installing all packages...",
                 bold=True,
             )
         )
-        # Uninstall all dev-packages from environment
-        for package in project.get_pipfile_section("packages"):
-            _uninstall_from_environment(project, package, system=system)
-        # Remove the package from the Pipfile
-        if project.reset_category_in_pipfile(category="packages"):
-            click.echo("Removed [packages] from Pipfile.")
+        # Uninstall all packages from all groups
+        for category in project.pipfile_sources:
+            if category in ["source", "requires"]:
+                continue
+            for package in project.get_pipfile_section(category):
+                _uninstall_from_environment(project, package, system=system)
+            # Remove the packages from the Pipfile
```

Comment:
I can't verify that this comment is accurate for what the code is doing, but removing the packages from the Pipfile is exactly what we don't want to do during uninstall --all to be consistent with the documented behavior.  

There was a test changed in a previous PR that should be reverted:  https://github.com/pypa/pipenv/pull/6029#discussion_r1674707464
If the test as it currently stands is passing, then uninstall --all isn't working like it used to.